Intoduce LAMMPS GUI and rewrite software requirements section (#5)
lammpstutorials/lammpstutorials-article: lammps-tutorials.tex

\begin{frontmatter}
\maketitle

\begin{abstract}
  The availability of open-source molecular simulation software packages
  allows scientists and engineers to focus on running and analyzing
  simulations without having to write, parallelize, and validate their
  own simulation software.  While molecular simulations thus become
  accessible to a larger audience, the ``black box'' nature of such
  software packages and wide array of options and features can make it
  challenging to use them correctly, particularly for beginners in the
  topic of MD simulations.  LAMMPS is one such versatile molecular
  simulation code, designed for modeling particle-based systems for a
  broad range of materials science and computational chemistry
  applications, including atomistic, coarse--grained, mesoscale,
  grid-free continuum, and discrete element models.  LAMMPS is capable
  of efficiently running simulations of varying sizes from small desktop
  computers to large-scale supercomputing environments.  Its flexibility
  and extensibility make it ideal for complex and extensive simulations

>   of atomic and molecular systems and beyond.  This article introduces a

Reviewer: ```suggestion
  of atomic and molecular systems, and beyond.  This article introduces a
```

  suite of tutorials designed to make learning LAMMPS more accessible to
  new users.  The first four tutorials cover the basics of running
  molecular simulations in LAMMPS with systems of varying complexities,
  including a simple fluid and a carbon nanotube.  The last three
  tutorials address more advanced molecular simulation techniques,

>   specifically the use of a reactive force field, enhanced sampling, and

Reviewer: specifically the use of a reactive force field, enhanced sampling, and
```suggestion
  specifically, the use of a reactive force field, enhanced sampling, and
```

ratio).  Some of these quantities can be directly compared with
experimental data, enabling the validation of the simulated system,
while others, available only through simulations, are often useful for
interpreting experimental data \cite{van2008molecular}.

LAMMPS (Large-scale Atomic/Molecular Massively Parallel Simulator)
\cite{lammps_home} is a highly flexible and parallel open-source MS
tool.  Over the years, a broad variety of particle interaction models
have been implemented in LAMMPS, enabling it to model -- among others --
atomic, polymeric, biological, metallic, reactive, granular, mesoscale,
grid--free continuum, and coarse-grained systems
\cite{thompson2022lammps}.  LAMMPS can be used on a single CPU core, a
multi-socket and multi-core server, an HPC cluster, or a high-end
supercomputing system.  It can handle complex and large-scale
simulations efficiently, including hybrid MPI--OpenMP parallelization
and MPI + GPU acceleration (for a subset of its functionality).

However, LAMMPS requires users to provide detailed input files that can
be challenging to write correctly, considering the overwhelming number

> of features and options offered by LAMMPS.  While LAMMPS has an

Reviewer: ```suggestion
of features and options offered by LAMMPS.  While LAMMPS has
```

extensive and detailed documentation, it can be difficult for new users
to navigate since it focuses on describing \emph{all} available
features, but expects that users already know which to choose.  Many of
the advanced features are not relevant to most users and the defaults
usually adequate and thus the level of detail in the manual can become
distracting.

In addition to the intrinsic complexity of LAMMPS, performing accurate
MS involves making several complex and system-specific decisions related
to the physics to be modeled, such as selecting the thermodynamic
ensemble (e.g. micro-canonical, grand-canonical), determining the
simulation duration, and choosing the sets of parameters describing the
interactions between the atoms (the so-called force field)

> \cite{van2018validation}.  These choices are independend from the chosen

Reviewer: ```suggestion
\cite{van2018validation}.  These choices are independent of the chosen
```

simulation software, although sometimes limited by the availability of
features in a given software package.  The tutorials presented in this
article aim to flatten the learning curve, and help users perform
accurate and reliable MS using LAMMPS.

\subsection{Scope}

This set of tutorials consists of seven tutorials arranged in order of
increasing difficulty.  The novelties associated with each tutorial are
briefly described below.

In \hyperref[lennard-jones-label]{tutorial 1}, the structure of LAMMPS

> input files is illustrated through creation of a simple, atomic

Reviewer: ```suggestion
input files is illustrated through the creation of a simple, atomic
```

Lennard-Jones fluid system.  Basic LAMMPS commands are used to set up
interactions between atoms, perform an energy minimization, and finally

> a simple MD simulation in the microcanonical (NVE) and canonical (NVT)

Reviewer: ```suggestion
run a simple MD simulation in the microcanonical (NVE) and canonical (NVT)
```

> ensembles.  This tutorial also introduce the LAMMPS graphical user

Reviewer: ```suggestion
ensembles.  This tutorial also introduces the LAMMPS graphical user
```

dedicated GitHub account,
\href{https://github.com/lammpstutorials}{github.com/lammpstutorials}. In
the following, all the commands that must be copied and pasted into
LAMMPS input files or the terminal are written in a \verb+verbatim+
style, and the file and folder names are written in \textit{italic}.

\subsection{Tutorial 1: Lennard-Jones fluid}
\label{lennard-jones-label}

The objective of this tutorial is to perform a simple MD simulation
using LAMMPS.  The system is a Lennard-Jones fluid composed of neutral
particles with two different effective diameters, contained within a
cubic box with periodic boundary conditions (Fig.\,\ref{fig:LJ}).  In
this tutorial, the temperature of the system is maintained using a
Langevin thermostat \cite{schneider1978molecular}, and basic quantities,
including the potential and kinetic energies, are extracted from the
system.

\subsubsection{My first input}

To run a simulation using LAMMPS, you need to write an input script with
series of commands of commands that LAMMPS will then execute.  For
clarity, the input scripts for this tutorial will be divided into five
categories, which we will fill out step by step.  First, create a folder

> named \textit{my-first-input/} for all files of this tutorial, then

Author: updated using "first part of this tutorial"

 # 2) System definition
 # 3) Settings
 # 4) Visualization
 # 5) Run
\end{verbatim}
}

A line or part of a line starting with a hash symbol ($\#$) is a
comment ignored by LAMMPS.  LAMMPS--GUI will color such comments in red.
These five categories are not required in every input script and do not
necessarily need to be in that exact order.  For instance, parts 3 and 4
could be inverted, or part 4 could be omitted.  Note, however, that
LAMMPS reads input files from top to bottom and processes each command
in it \emph{immediately}.  Therefore, the \textit{Initialization} and
\textit{System definition} categories must appear at the top of the
input, and the \textit{Run} category at the bottom.  Also, the specifics
of some commands can change after global settings are changed, so the
order of commands in the input script matters for such cases.

The contents of the \textit{Editor} window can be saved by either using
the ``Save'' or ``Save As'' entry in the ``File'' menu, using the
\texttt{Ctrl--S} keyboard shortcut or by clicking on the ``Save'' icon
at the bottom left of the \textit{Editor} window status bar.  For
running LAMMPS in LAMMPS--GUI it is not required to save the buffer.

> The current contents of the buffer will be passed on to LAMMPS.

Author: One of the earliest design choices for LAMMPS-GUI was to run directly from the content of the edit buffer, i.e. to get away from the "save then run" setup required with other editors. The rationale being that LAMMPS-GUI would be used for quick experimenting. Since almost everything can be done with keyboard shortcuts, saving the content of the buffer can be quick (Ctrl-S and Ctrl-Enter) will do what you expect.

But I can see how this would be unexpected for others. I don't want to change the default behavior, but I think an "auto-save" could be added as an optional feature. One could provide an "idle interval" or save on "run" or "quit". I'll add this to my TODO list.

> The current contents of the buffer will be passed on to LAMMPS.

Author: FYI, an auto-save or "run" and "quit" has been implemented in commit https://github.com/lammps/lammps/pull/4255/commits/d470919a4424e4cb618b4df9560a442ae6f909d3
I need to update the docs and then will push new binaries to the testing download server tonight.

If the plot in your \textit{Chart} window is missing data, then your
computer is running the simulation too fast for LAMMPS--GUI to collect
all the data.  You can adjust the settings in the
\textit{Edit~$\to$~Preferences} dialog to collect data more frequently
by reducing the \textit{GUI update interval} value.  At values lower
than 10ms the time the GUI spends on collecting the data becomes
significant and thus those should only be used for fast tutorials like
this one.  You can also make your simulation slower by selecting
\textit{None} in the \textit{Accelerators} tab.

The potential energy can be seen to decrease from a positive value to a
negative value.  The initially positive value of the potential energy is
expected because the atoms have been created at random positions within
the simulation box and some of them are very close, resulting in a large
initial potential energy due to the repulsive branch of the
Lennard-Jones potential.  As the energy minimization progresses, the
energy decreases -- first rapidly -- then more slowly and plateaus at
a negative value, indicating that the atoms have been displaced to
reasonable distances from each other.

> Since the \textit{thermo\_style} command also included the output of the

Author: writing "ed" instead of "es" is a common typo I do. I suspect this is due to having never properly learned to type "blind". It happens because "d" is right next to "s" on my keyboard. ;-)

they differ) \cite{schneider1978molecular}.  A \textit{damping}
parameter of 0.1 is used.  The \textit{damping} parameter determines how
rapidly the temperature is relaxed to its desired value.  In a Langevin
thermostat the atoms are subject friction and random noise (in the form
of randomly added velocities).  Since a constant friction term removes
more kinetic energy from fast atoms and less from slow atoms, the system
will eventually reach a dynamic equilibrium where the kinetic energy
removed and added are about the same.  The number \textit{1530917} is a
seed used to initialize the random number generator used inside of
\textit{fix langevin}, you can change it to perform statistically
independent simulations.

Run the simulation again using LAMMPS. Open the \textit{Output} window
by selecting \textit{View}, and \textit{Output}. From the information
printed in the \textit{Output} window, one can see that the temperature
\textit{Temp} starts from 0, but rapidly reaches the requested value and
stabilizes itself near $T=1$ temperature units.  One can also see that
the potential energy, $p_\text{e}$, rapidly decreases during energy

> minimization (see also Fig.\,\ref{fig:evolution-energy}\,a). Then, after

Reviewer: ```suggestion
minimization (see also Fig.\,\ref{fig:evolution-energy}\,a). After
```
